robusta/src/ws_message.rs

```diff
@@ -1,24 +1,52 @@
 use serde::{Deserialize, Serialize};
 
+use crate::gadgets::{DetectiveGadget, MrXGadget};
+
 #[derive(specta::Type, Clone, Deserialize, Debug)]
 pub enum ClientMessage {
     Position { long: f32, lat: f32 },
     SetTeamPosition { long: f32, lat: f32 },
     JoinTeam { team_id: u32 },
     EmbarkTrain { train_id: String },
     DisembarkTrain,
+    MrXGadget(MrXGadget),
+    DetectiveGadget(DetectiveGadget),
     Message(String),
 }
 
 #[derive(specta::Type, Clone, Serialize, Deserialize, Debug)]
 pub enum ClientResponse {
     GameState(GameState),
+    MrXGadget(MrXGadget),
+    DetectiveGadget(DetectiveGadget),
+    MrXPosition(MrXPosition),
```

Comment:
If we add more gadget this might become cluttered, thus you might want to wrap this in a `UseGadget(Gadget)`